api/v1alpha1/clientprofile_types.go

```diff
@@ -31,6 +31,9 @@ type CephFsConfigSpec struct {
 
 	//+kubebuilder:validation:Optional
 	FuseMountOptions map[string]string `json:"fuseMountOptions,omitempty"`
+
+	//+kubebuilder:validation:Optional
+	RadosNamespace string `json:"radosNamespace,omitempty"`
```

Comment:
in the linked PR, it was mentioned this field should be added to newer deployments only, should we support that or will there be any way to migrate the older way to use rns for omap data?

Reply:
> in the linked PR, it was mentioned this field should be added to newer deployments only, should we support that 

```
# NOTE: Make sure you don't add radosNamespace option to a currently in use
# configuration as it will cause issues.
```

Yes, the radosNamespace field should only be included in new clientProfile configuration to avoid potential issues. 
Users should note that they don't add the radosNamespace option to an existing clientProfile configuration, as it may disrupt current operations.
Since Ceph-CSI supports specifying a radosNamespace for CephFS volumes, the Ceph-CSI operator should also support it.  Thoughts @Madhu-1?

> will there be any way to migrate the older way to use rns for omap data?

At this time, no migration design exists for migrating OMAP data between namespaces.

Reply:
No migration is supported from cephcsi side. this field should be immutable and should not be changed upon creation